services/QuillLMS/app/models/unit_activity.rb

```diff
@@ -157,6 +157,11 @@ def self.get_classroom_user_profile(classroom_id, user_id)
           COALESCE(cuas.locked, false) AS locked,
           COALESCE(cuas.pinned, false) AS pinned,
           MAX(acts.percentage) AS max_percentage,
+          unit.visible AS unit_visible,
+          ua.visible AS ua_visible,
+          cu.visible AS cu_visible,
+          acts.visible AS acts_visible,
+          CASE WHEN unit.visible = false OR ua.visible = false OR cu.visible = false OR acts.visible = false THEN true ELSE false END as archived,
```

Comment:
I'm not sure that this is actually easier to read, but because we're always looking for the same value (`false`), this can be:
```sql
CASE WHEN false IN (unit.visible, ua.visible, cu.visible, acts.visible) THEN true ELSE false END as archived
```